CMakeLists.txt

```diff
@@ -1,179 +1,141 @@
+# Include libraries
+add_subdirectory(${LIBS_DIR}/actuators)
+add_subdirectory(${LIBS_DIR}/can)
+add_subdirectory(${LIBS_DIR}/encoder)
+add_subdirectory(${LIBS_DIR}/gamepad)
+add_subdirectory(${LIBS_DIR}/gpio)
+add_subdirectory(${LIBS_DIR}/controllers)
+add_subdirectory(${LIBS_DIR}/neopixel)
+add_subdirectory(${LIBS_DIR}/pid)
+add_subdirectory(${LIBS_DIR}/sensors)
+add_subdirectory(${LIBS_DIR}/utility)
+
+# Only add subdirectory of requested APP
+function(add_app_subdirectory APP_SUBDIRECTORY)
+    string(FIND ${APP_SUBDIRECTORY} / INDEX REVERSE)
+    math(EXPR INDEX "${INDEX} + 1")
+    string(SUBSTRING ${APP_SUBDIRECTORY} ${INDEX} -1 APP_NAME)
+
+    if ("${APP_NAME}" STREQUAL "")
+        message(FATAL_ERROR "add_app_subdirectory could not extract an app name from ${APP_SUBDIRECTORY}")
+    elseif ("$ENV{APP}" STREQUAL ${APP_NAME})
+        add_subdirectory(${APP_SUBDIRECTORY})
+    endif ()
+endfunction()
+
+## Include Rover Apps
+add_app_subdirectory(${APPS_DIR}/arm)
+add_app_subdirectory(${APPS_DIR}/gamepad)
+add_app_subdirectory(${APPS_DIR}/gimbal)
+add_app_subdirectory(${APPS_DIR}/pdb)
+add_app_subdirectory(${APPS_DIR}/science)
+
+# Include Test Apps
+add_app_subdirectory(${TEST_APPS_DIR}/test-actuator-controller)
+add_app_subdirectory(${TEST_APPS_DIR}/test-adafruitSTEMMA)
+add_app_subdirectory(${TEST_APPS_DIR}/test-aeat-6012)
+add_app_subdirectory(${TEST_APPS_DIR}/test-aeat-8800)
+add_app_subdirectory(${TEST_APPS_DIR}/test-blinky)
+add_app_subdirectory(${TEST_APPS_DIR}/test-blockingneo)
+add_app_subdirectory(${TEST_APPS_DIR}/test-can)
+add_app_subdirectory(${TEST_APPS_DIR}/test-mae3)
+add_app_subdirectory(${TEST_APPS_DIR}/test-logger)
+add_app_subdirectory(${TEST_APPS_DIR}/test-lookup-table)
+add_app_subdirectory(${TEST_APPS_DIR}/test-pid)
+add_app_subdirectory(${TEST_APPS_DIR}/test-pwm)
+add_app_subdirectory(${TEST_APPS_DIR}/test-pwmin)
+add_subdirectory(${TEST_APPS_DIR}//test-units)
+
+# Ensure APP was specified
+if ("$ENV{APP}" STREQUAL "")
+    message(FATAL_ERROR "APP was not specified")
+endif()
+
+# Ensure that a target exists for the specified APP
+# https://stackoverflow.com/a/62311397/10173605
+function(get_all_targets var)
+    set(targets)
+    get_all_targets_recursive(targets ${CMAKE_CURRENT_SOURCE_DIR})
+    set(${var} ${targets} PARENT_SCOPE)
+endfunction()
+
+macro(get_all_targets_recursive targets dir)
+    get_property(subdirectories DIRECTORY ${dir} PROPERTY SUBDIRECTORIES)
+    foreach (subdir ${subdirectories})
+        get_all_targets_recursive(${targets} ${subdir})
+    endforeach ()
+
+    get_property(current_targets DIRECTORY ${dir} PROPERTY BUILDSYSTEM_TARGETS)
+    list(APPEND ${targets} ${current_targets})
+endmacro()
+
+get_all_targets(ALL_BUILDSYSTEM_TARGETS)
+if ("$ENV{APP}" IN_LIST ALL_BUILDSYSTEM_TARGETS)
+    message(STATUS "$ENV{APP} app is being built")
+else ()
+    message(FATAL_ERROR "$ENV{APP} app is not an app target found in ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt")
+endif ()
```

Comment:
This fancy chunk of cmake checks the list of ALL cmake targets to see if there's a target's whos name matches the  `APP=` that was passed in